library/network_connections.py

```diff
@@ -813,6 +825,11 @@ def connection_create(self, connections, idx, connection_current=None):
             s_infiniband.set_property(
                 NM.SETTING_INFINIBAND_MAC_ADDRESS, connection["mac"]
             )
+            if hasattr(NM, "SettingMatch") and hasattr(NM, "SETTING_MATCH_PATH"):
```

Comment:
this code is inside

```python
if connection["type"] == "ethernet":
    ...
elif connection["type"] == "infiniband":
    ...
```

The match option is independent of the connection type (or if not, `ArgValidator` should check that).

Hence, this code should not be duplicated inside these `if`s.

Reply:
also, I don't think that the role should add a `[match]` section, if there are no interface_path parameters requested. Maybe that was the reason why you added the code inside these blocks?

Together with dropping `default_value=list`, maybe just do:
```python
   if connection["interface_path"] is not None:
       s_match = self.connection_ensure_setting(con, NM.SettingMatch)
       s_match.set_property(
             NM.SETTING_MATCH_PATH, connection["interface_path"]
       )
```

Reply:
Updated, see
https://github.com/linux-system-roles/network/blob/aa27d56c6d67d7b883b5415a3a972678642ab835/library/network_connections.py#L1115